core/src/main/java/org/zenoss/zep/zing/impl/ZingEventProcessorImpl.java

```diff
@@ -76,60 +113,67 @@ public boolean enabled() {
-            builder.setUuid(uuid);
-            if (event.hasFingerprint()) builder.setFingerprint(event.getFingerprint());
-            if (event.hasSeverity()) builder.setSeverity(event.getSeverity().name());
-            EventActor actor = event.getActor();
-            if (actor != null) {
-                if (actor.hasElementUuid()) builder.setParentContextUUID(actor.getElementUuid());
-                if (actor.hasElementIdentifier()) builder.setParentContextIdentifier(actor.getElementIdentifier());
-                if (actor.hasElementTitle()) builder.setParentContextTitle(actor.getElementTitle());
-                if (actor.hasElementTypeId()) builder.setParentContextType(actor.getElementTypeId().name());
-                if (actor.hasElementSubUuid()) builder.setContextUUID(actor.getElementSubUuid());
-                if (actor.hasElementSubIdentifier()) builder.setContextIdentifier(actor.getElementSubIdentifier());
-                if (actor.hasElementSubTitle()) builder.setContextTitle(actor.getElementSubTitle());
-                if (actor.hasElementSubTypeId()) builder.setContextType(actor.getElementSubTypeId().name());
-            }
-            if (event.hasMessage()) builder.setMessage(event.getMessage());
-            if (event.hasSummary()) builder.setSummary(event.getSummary());
-            if (event.hasMonitor()) builder.setMonitor(event.getMonitor());
-            if (event.hasAgent()) builder.setAgent(event.getAgent());
-            if (event.hasEventKey()) builder.setEventKey(event.getEventKey());
-            if (event.hasEventClass()) builder.setEventClass(event.getEventClass());
-            if (event.hasEventClassKey()) builder.setEventClassKey(event.getEventClassKey());
-            if (event.hasEventClassMappingUuid()) builder.setEventClassMappingUuid(event.getEventClassMappingUuid());
-            if (event.hasEventGroup()) builder.setEventGroup(event.getEventGroup());
-            if (summary.hasCount()) builder.setCount(summary.getCount());
-            if (summary.hasFirstSeenTime()) builder.setFirstSeen(summary.getFirstSeenTime());
-            if (summary.hasLastSeenTime()) builder.setLastSeen(summary.getLastSeenTime());
-            if (summary.hasUpdateTime()) builder.setUpdateTime(summary.getUpdateTime());
-            if (summary.hasStatus()) builder.setStatus(summary.getStatus().name());
-            if (summary.hasClearedByEventUuid()) builder.setClearedByUUID(summary.getClearedByEventUuid());
-            for (EventDetail d : event.getDetailsList()) {
-                builder.setDetail(d.getName(), d.getValueList());
-            }
-
-            ZingEvent zingEvent = builder.build();
-            logger.debug("publishing event {}", zingEvent);
-            if (zingEvent.isValid()) {
-                this.publisher.publishEvent(zingEvent);
-            } else {
-                this.invalidEventsCounter.inc();
+            int eventSev = summary.getOccurrence(0).getSeverity().getNumber();
```

Comment:
Consider using this.minSeverity.compareTo(summary.getOccurrence(0).getSeverity())